src/python/grpcio/grpc/_common.py

```diff
@@ -62,22 +62,30 @@
 }
 
 
-def encode(s):
-    if isinstance(s, bytes):
+def encode(s):  #pylint: disable=inconsistent-return-statements
+    if isinstance(s, bytes) or (six.PY2 and isinstance(s, str)):
         return s
     else:
-        return s.encode('ascii')
+        for encoding in ['utf8', 'latin1']:
+            try:
+                return s.encode(encoding)
+            except UnicodeEncodeError:
+                pass
+        _LOGGER.exception('Cannot encode %s', s)
+        return s
 
 
-def decode(b):
-    if isinstance(b, str):
+def decode(b):  #pylint: disable=inconsistent-return-statements
```

Comment:
maybe smth simple like this?
```
if isinstance(b, bytes):
  return b.decode('utf-8', 'replace')
return b
```

similar for `def encode`, just always encoding it as `utf-8` should make a deal

Reply:
If possible, I want to keep the message content. The replace will get rid of character that `utf-8` can't understand. It will fail some cases, if people transmit pure binary data.

Reply:
gRPC HTTP/2 spec explicitly states that message is an utf-8 message, so pure binary data is not an acceptable input (and therefore it should be sanitized, but not honored).
Please see https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#responses for more details.

It will be also consistent to grpc-go behavior, which currently sanitizes status messages by replacing invalid bytes sequences with the Unicode replacement character.

Reply:
Please review the latest change. I have adopted your advices and added test cases.